### PR TITLE
add missing methods and fix tests

### PR DESCRIPTION
**Pull request by dpo**
supersedes #10

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -8,14 +8,14 @@ FastClosures = "9aa1b823-49e4-5ca5-8b0f-3971ec8bab6a"
...
 
 [compat]
-FastClosures = "≥ 0.2.1"
-LinearOperators = "0.5.0"
-NLPModels = "0.6.1, < 0.9.0"
-Requires = "≥ 0.3.0"
+FastClosures = "0.2.1, 0.3.0"
+LinearOperators = "0.7.0, 1"
+NLPModels = "0.10.0, 1"
+Requires = "0.3.0, 0.5"
```
Comment (abelsiqueira) [inline Project.toml]: There is no range in Pkg, yet. This needs to be "0.3, 0.4, 0.5".

Comment (dpo) [reply Project.toml]: Isn't [this](https://julialang.github.io/Pkg.jl/v1/compatibility/#Version-specifier-format-1) a range?

Comment (abelsiqueira) [reply Project.toml]: No, that's a union of two `caret` specifiers: `1.2` implies `[1.2,2.0)` and `2` implies `[2,3)`, which together form the `[1.2.0,0.3.0)` specified.

Comment (dpo) [reply Project.toml]: How are the LinearOperators and FastClosures valid then? I obviously understand nothing that that versioning system, and the docs don't help.

Comment (abelsiqueira) [reply Project.toml]: FastClosures works for `[0.2.1,0.4.0)`, and LinearOperators works for `[0.7,0.8) U [1.0,2.0)`. I think it lacks examples, especially in situations where it differs from ranged conditions. But from what I understood, there are only `~` and `^` (and inequalities, but ignore those?).  `^` means up to, not including, the next semver, and `~` is a more limited version of it. On pre-1.0 releases, the next semver of `0.x.y` is `0.(x+1)`.

**Comment by abelsiqueira** [comment]
Thank you

**Pull request merged**